src/front/spv/null.rs

```diff
@@ -0,0 +1,149 @@
+                let inner = generate_null_constant(member_ty, type_arena, constant_arena)?;
+                components.push(constant_arena.fetch_or_append(crate::Constant {
+                    name: None,
+                    specialization: None,
+                    inner,
+                }));
+            }
+            crate::ConstantInner::Composite { ty, components }
+        }
+        crate::TypeInner::Array {
+            base,
+            size: crate::ArraySize::Constant(handle),
+            ..
+        } => {
+            let size = constant_arena[handle]
+                .to_array_length()
+                .ok_or(Error::InvalidArraySize(handle))?;
+            let inner = generate_null_constant(base, type_arena, constant_arena)?;
+            let value = constant_arena.fetch_or_append(crate::Constant {
+                name: None,
+                specialization: None,
+                inner,
+            });
+            crate::ConstantInner::Composite {
+                ty,
+                components: vec![value; size as usize],
+            }
+        }
+        ref other => {
+            log::warn!("null constant type {:?}", other);
+            return Err(Error::UnsupportedType(ty));
+        }
+    };
+    Ok(inner)
+}
+
+/// Create a default value for an output built-in.
+pub fn generate_default_built_in(
+    built_in: Option<crate::BuiltIn>,
+    ty: Handle<crate::Type>,
+    type_arena: &mut Arena<crate::Type>,
+    constant_arena: &mut Arena<crate::Constant>,
+) -> Result<Handle<crate::Constant>, Error> {
+    let inner = match built_in {
+        Some(crate::BuiltIn::Position) => {
+            let zero = constant_arena.fetch_or_append(crate::Constant {
+                name: None,
+                specialization: None,
+                inner: crate::ConstantInner::Scalar {
+                    value: crate::ScalarValue::Float(0.0),
+                    width: 4,
+                },
+            });
+            let one = constant_arena.fetch_or_append(crate::Constant {
+                name: None,
+                specialization: None,
+                inner: crate::ConstantInner::Scalar {
+                    value: crate::ScalarValue::Float(1.0),
+                    width: 4,
+                },
+            });
+            crate::ConstantInner::Composite {
+                ty,
+                components: vec![zero, zero, zero, one],
+            }
+        }
+        Some(crate::BuiltIn::PointSize) => crate::ConstantInner::Scalar {
+            value: crate::ScalarValue::Float(1.0),
+            width: 4,
+        },
+        Some(crate::BuiltIn::FragDepth) => crate::ConstantInner::Scalar {
+            value: crate::ScalarValue::Float(0.0),
+            width: 4,
+        },
+        Some(crate::BuiltIn::SampleMask) => crate::ConstantInner::Scalar {
+            value: crate::ScalarValue::Uint(!0),
+            width: 4,
+        },
+        //Some(crate::BuiltIn::ClipDistance) |
```

Comment:
nit: leftover comment

Reply:
this is actually by design, I'll add more wording